google-cloud-core/src/main/java/com/google/cloud/BaseService.java

```diff
@@ -51,6 +51,8 @@ public RetryResult beforeEval(Exception exception) {
   public static final ExceptionHandler EXCEPTION_HANDLER =
       ExceptionHandler.newBuilder()
           .abortOn(RuntimeException.class)
+          .retryOn(java.net.ConnectException.class)
+          .retryOn(java.net.UnknownHostException.class)
```

Comment:
When there's such errors, I don't think retrying would resolve the underlying network problems.